src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -155,6 +155,19 @@ internal static void ThrowNegateTwosCompOverflow()
             throw new OverflowException(SR.Overflow_NegateTwosCompNum);
         }
 
+        internal static unsafe ulong BigMul(uint a, uint b)
+        {
+#if TARGET_32BIT
+            if (Bmi2.IsSupported)
+            {
+                uint low;
+                uint high = Bmi2.MultiplyNoFlags(a, b, &low);
```

Comment:
Did you measure the performance for this? It's very likely to be significantly worse than the previous codegen that the JIT automatically generated for uint*uint->ulong multiplication on x86.

Reply:
Sorry no i didn't know how to do benchmarks. Probably we should just remove the intrinsic block for x86 in BigMul methods if they are slow? Same for 64 x 64 = 128

Reply:
There's an existing issue for MULX coden quality: #11782
And a different related issue: #75594

Reply:
And a 3rd issue that is very closely related to this `DecCalc` code: #58263